### PR TITLE
Added filters flag

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR

- Added flyteidl auto-release automation
- Added flags for filter, sort-by, and limit flags. 
- Refactor list command. In past we are using GetEntity API call
- Refactor sub command flags 
- Added test for more coverage 

```
$ flytectl  get projects --filter.field-selector="project.Value>4,project.Phase contains FAIL"
$ flytectl  get projects  --filter.limit=1 --filter.sorty-by=filter.created_at --filter.asc

```

```
- contains - --filter.field-selector="project.Phase contains FAIL" 
- gt (greater than) - --filter.field-selector="project.Value>4"
- gte (greter than or equal to) - --v.field-selector="project.Value>=4"
- lt (less than) - --filter.field-selector="project.Value<5"
- lte (less than or equal to) - --filter.field-selector="project.Value<=5"
- eq (equal) - --filter.field-selector="project.name=X"
- ne (not equal) - --filter.field-selector="project.name!=X"
- value_in (for repeated sets of values) - --filter.field-selector="project.Phase in (RUNNING;FAILED)" 
```

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/736

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Review by kumare3 — APPROVED** [review]

**Comment by evalsocket** [comment]
@kumare3 let's wait for @pmahindrakar-oss +1. we can merge it tomorrow

**Comment by pmahindrakar-oss** [comment]
I had already approved it in previous branch. This need not have waited though.Anyways merging this in

**Review by pmahindrakar-oss — APPROVED** [review]
